addons/product/models/product_product.py

```diff
@@ -631,7 +631,9 @@ def action_open_documents(self):
     #=== BUSINESS METHODS ===#
 
     def _prepare_sellers(self, params=False):
-        return self.seller_ids.filtered(lambda s: s.partner_id.active).sorted(lambda s: (s.sequence, -s.min_qty, s.price, s.id))
+        potential_sellers = self.seller_ids.filtered(lambda s: s.partner_id.active)
+        sellers = potential_sellers.filtered(lambda seller: not seller.product_id or seller.product_id == self) or potential_sellers
```

Comment:
In case you have only one seller that replenish only one variant, `or potential_sellers` will take it for any variant. To me it's an error that we should block

Reply:
Agreed !